src/pd/client.rs

```diff
@@ -92,12 +98,15 @@ pub trait PdClient: Send + Sync + 'static {
                 Some(sk) => sk,
             };
             let end_key = end_key.clone();
-
             let this = self.clone();
             Either::Left(self.region_for_key(&start_key).and_then(move |region| {
                 let region_end = region.end_key();
                 this.map_region_to_store(region).map_ok(move |store| {
-                    if end_key.map(|x| x <= region_end).unwrap_or(false) || region_end.is_empty() {
+                    if end_key
+                        .map(|x| x <= region_end && !x.is_empty())
```

Comment:
Not sure whether the condition `!x.is_empty()` is necessary

Reply:
Yes I think it's necessary